sdk/communication/azure-communication-callingserver/tests/test_call_connection_async.py

```diff
@@ -0,0 +1,1064 @@
+@pytest.mark.asyncio
+async def test_cancel_all_media_operations_failed(
+    test_name, # type: str
+    call_connection_id, # type: str
+    operation_context = None, # type: str
+    use_managed_identity = False # type: bool
+    ):
+
+    call_connection = _mock_utils_async.create_mock_call_connection(
+        call_connection_id,
+        status_code=404,
+        payload=_test_constants.ErrorPayload,
+        use_managed_identity = use_managed_identity
+        )
+
+    raised = False
+    try:
+        await call_connection.cancel_all_media_operations(operation_context)
+    except:
+        raised = True
+    assert raised == True
+
+@parameterized.expand(CallConnectionUnitTestUtils.data_source_test_play_audio())
+@pytest.mark.asyncio
+async def test_play_audio_succeed(
+    test_name, # type: str
+    call_connection_id, # type: str
+    audio_url, # type: str
+    is_looped, # type: bool
+    audio_file_id, # type: str
+    operation_context, # type: str
+    use_managed_identity = False # type: bool
+    ):
+
+    call_connection = _mock_utils_async.create_mock_call_connection(
+        call_connection_id,
+        status_code=202,
+        payload=_test_constants.PlayAudioResponsePayload,
+        use_managed_identity=use_managed_identity
+        )
+
+    result = await call_connection.play_audio(
+        audio_url,
+        is_looped,
+        audio_file_id = audio_file_id,
+        operation_context = operation_context
+        )
+    CallConnectionUnitTestUtils.verify_play_audio_result(result)
+
+@parameterized.expand(CallConnectionUnitTestUtils.data_source_test_play_audio())
+@pytest.mark.asyncio
+async def test_play_audio_failed(
+    test_name, # type: str
+    call_connection_id, # type: str
+    audio_url, # type: str
+    is_looped, # type: bool
+    audio_file_id, # type: str
+    operation_context, # type: str
+    use_managed_identity = False # type: bool
+    ):
+
+    call_connection = _mock_utils_async.create_mock_call_connection(
+        call_connection_id,
+        status_code=404,
+        payload=_test_constants.ErrorPayload,
+        use_managed_identity = use_managed_identity
+        )
+
+    raised = False
+    try:
+        await call_connection.play_audio(
+            audio_url,
+            is_looped,
+            audio_file_id = audio_file_id,
+            operation_context = operation_context
+            )
+    except:
+        raised = True
+    assert raised == True
+
+@parameterized.expand(CallConnectionUnitTestUtils.data_source_test_add_participant())
+@pytest.mark.asyncio
+async def test_add_participant_succeed(
+    test_name, # type: str
+    call_connection_id, # type: str
+    participant, # type: CommunicationIdentifier
+    alternate_caller_id = None, # type: PhoneNumberIdentifier
+    operation_context = None, # type: str
+    use_managed_identity = False # type: bool
+    ):
+
+    call_connection = _mock_utils_async.create_mock_call_connection(
```

Comment:
What product code do these tests hit? It looks like you are creating a mock call_connection, then calling a method on the mock, and validating the response from the mock. I'm probably missing how this mock framework functions.

Reply:
If the funciton is returning a real call_connection that has mocked dependencies, then we should change the name.